### PR TITLE
New package: py-cupy

### PR DESCRIPTION
**Pull request by qwertos**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-cupy/package.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    """CuPy is an open-source array library accelerated with
+    NVIDIA CUDA. CuPy provides GPU accelerated computing with
+    Python. CuPy uses CUDA-related libraries including cuBLAS,
+    cuDNN, cuRand, cuSolver, cuSPARSE, cuFFT and NCCL to make
+    full use of the GPU architecture."""
+
+    homepage = "https://cupy.dev/"
+    git      = "https://github.com/cupy/cupy"
+
+    version('8.0.0', tag='v8.0.0', submodules=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cupy/package.py]: Git checkouts aren't checksummable. Would it be possible to compile from the release tarballs on PyPI, or are the submodules required?

Comment (qwertos) [reply var/spack/repos/builtin/packages/py-cupy/package.py]: Submodules are required. I don't know if they are included in the pypi tarballs.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-cupy/package.py]: Can you check?

File: var/spack/repos/builtin/packages/py-cupy/package.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    Python. CuPy uses CUDA-related libraries including cuBLAS,
+    cuDNN, cuRand, cuSolver, cuSPARSE, cuFFT and NCCL to make
+    full use of the GPU architecture."""
+
+    homepage = "https://cupy.dev/"
+    url = "https://pypi.io/packages/source/c/cupy/cupy-8.0.0.tar.gz"
+
+    version('8.0.0', sha256='d1dcba5070dfa754445d010cdc952ff6b646d5f9bdcd7a63e8246e2472c3ddb8')
+
+    depends_on('python@3.6:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cupy/package.py:21]:
```suggestion
    depends_on('python@3.5:', type=('build', 'run'))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-cupy/package.py (hunk 2 above)
Comment (qwertos) [reply var/spack/repos/builtin/packages/py-cupy/package.py:21]: They don't advise that and this requires setting an environment var to override the safety. Is this necessary?

https://github.com/cupy/cupy/blob/36aa7523edd203f0692845495e9bca0f7b5702fa/setup.py#L24-L33

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-cupy/package.py:21]: Oh right, you're correct.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**